### PR TITLE
User can register and login with Facebook

### PR DESCRIPTION
**Pull request by GergKllai1**
# PT Link

[User can sign in using Facebook](https://www.pivotaltracker.com/story/show/163530690)

# User Story

```
As a user
In order to gain access to Tribal's features
I would like to be able to sign in using Facebook
```

# Screenshots

feb-02-2019 17-08-33

# Description of PR

- Implements facebook auth gem in rails backend
- Adds register and login links in react component
- Sends down role params during registration

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -2,7 +2,30 @@ class User < ApplicationRecord
...
+
+	def self.from_omniauth(auth, params)
+		where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+			user.email = auth.info.email
+			user.first_name = auth.info.first_name
+			user.last_name = auth.info.last_name
+			if params.present?
+				user.role = params['role']
+			end
+      user.password = Devise.friendly_token[0,20]
```
Comment (HannaSwDn) [inline app/models/user.rb:26]: What does [0, 20] do?

File: app/models/user.rb [hunk 2]
```diff
@@ -2,7 +2,30 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+				 :recoverable, :rememberable, :validatable,
+				 :omniauthable, omniauth_providers: [:facebook]
+
```
Comment (HannaSwDn) [inline app/models/user.rb:7]: remove this line please :)

File: client/bundles/Main/components/Main.jsx [hunk 3]
```diff
@@ -34,6 +39,7 @@ export default class Main extends React.Component {
 			console.log(error);
```
Comment (HannaSwDn) [inline client/bundles/Main/components/Main.jsx:39]: do we need this console.log here?

File: app/models/user.rb (hunk 2 above)
Comment (kianaditya) [inline app/models/user.rb:7]: Remove extra empty lines

File: app/models/user.rb (hunk 1 above)
Comment (GergKllai1) [reply app/models/user.rb:26]: It's a devise method to create a token as password

File: client/bundles/Main/components/Main.jsx (hunk 3 above)
Comment (GergKllai1) [reply client/bundles/Main/components/Main.jsx:39]: Yes we need to console log errors in http requests.

File: app/models/user.rb (hunk 2 above)
Comment (GergKllai1) [reply app/models/user.rb:7]: Done

Comment (GergKllai1) [reply app/models/user.rb:7]: Done

**Review by kianaditya — APPROVED** [review]

**Review by HannaSwDn — APPROVED** [review]

**Review by schneiderman805 — COMMENTED** [review]
looks good to me

**Pull request merged**